src/services/dumpers/agent-nodejs.ts

```diff
@@ -0,0 +1,209 @@
+import type { Config, DbConfig } from '../../interfaces/project-create-interface';
+import type Lodash from 'lodash';
+
+import AbstractDumper from './abstract-dumper';
+
+export default class AgentNodeJs extends AbstractDumper {
+  private env: { FOREST_SERVER_URL: string; FOREST_URL_IS_DEFAULT: boolean };
+
+  private readonly DEFAULT_PORT = 3310;
+
+  private readonly isLinuxOs: boolean;
+
+  private readonly isDatabaseLocal: (dbConfig: DbConfig) => boolean;
+
+  private readonly buildDatabaseUrl: (dbConfig: DbConfig) => string;
+
+  private readonly lodash: typeof Lodash;
+
+  private readonly toValidPackageName: (string: string) => string;
+
+  constructor(context) {
+    const {
+      assertPresent,
+      env,
+      isLinuxOs,
+      buildDatabaseUrl,
+      isDatabaseLocal,
+      lodash,
+      toValidPackageName,
+    } = context;
+
+    assertPresent({
+      env,
+      isLinuxOs,
+      buildDatabaseUrl,
+      isDatabaseLocal,
+      lodash,
+      toValidPackageName,
+    });
+
+    super(context);
+
+    this.env = env;
+    this.isLinuxOs = isLinuxOs;
+    this.buildDatabaseUrl = buildDatabaseUrl;
+    this.isDatabaseLocal = isDatabaseLocal;
+    this.lodash = lodash;
+    this.toValidPackageName = toValidPackageName;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  protected get templateFolder() {
+    return 'agent-nodejs';
+  }
```

Comment:
I'll answer here since I can't comment on the first review comment.
"I'm wondering if we must keep only the FOREST_SERVER_URL en var inside the context. One var for one thing.
If we going into this we can consider A complete rename of this var inside a dedicated PR and juste keep here the addition of process.env.FOREST_SERVER_URL on the var declaration"

We made a mistake by introducing FOREST_SERVER_URL, we now have two variables to do the same thing. We can't just remove one of them now, because that would mean that we need to display the right env variable depending on the agent version during deployment.

Anyway, it's out of the scope of this epic IMO, and adding it to the context means we can use both without mistakes.

To be clear I agree that we should only have one.

See [here](https://forestadmin.slack.com/archives/GAZF5Q5RV/p1677592915838719) for a discussion on the matter.